src/py/mat3ra/made/tools/build/slab/configuration.py

```diff
@@ -1,42 +1,248 @@
+        )
+
+        return raw_slabs
+
+    def get_terminations(self):
+        return [Termination.from_string(label_pymatgen_slab_termination(slab)) for slab in self._pymatgen_slabs]
+
+    def get_slabs(
+        self,
+        min_slab_size: Union[float, int],
+        min_vacuum_size: Union[float, int],
+        in_unit_planes: bool,
+        make_primitive: bool,
+        symmetrize: bool,
+        use_orthogonal_c: bool,
+    ) -> List[Material]:
+        pymatgen_slabs = self._generate_pymatgen_slabs(
+            min_slab_size=min_slab_size,
+            min_vacuum_size=min_vacuum_size,
+            in_unit_planes=in_unit_planes,
+            make_primitive=make_primitive,
+            symmetrize=symmetrize,
+        )
+        orthogonal_slabs = (
+            [slab.get_orthogonal_c_slab() for slab in pymatgen_slabs] if use_orthogonal_c else pymatgen_slabs
+        )
+        material_slabs = [Material.create(from_pymatgen(slab)) for slab in orthogonal_slabs]
+        wrapped_material_slabs = [wrap_to_unit_cell(material) for material in material_slabs]
+        return wrapped_material_slabs
+
+
+class AtomicLayers(AtomicLayersSchema, CrystalLatticePlanes):
+    pass
+
+
+class AtomicLayersUnique(AtomicLayersUniqueSchema, CrystalLatticePlanes):
+    pass
+
+
+class AtomicLayersUniqueRepeated(AtomicLayersUniqueRepeatedSchema, CrystalLatticePlanes):
+    crystal: Material
+    termination_top: Termination
+
+
+class VacuumConfiguration(VacuumConfigurationSchema):
     """
-    Configuration for building a slab.
+    Configuration for adding vacuum to a material.
 
     Args:
-        bulk (Material): The bulk material.
-        miller_indices (Tuple[int, int, int]): The Miller indices of the slab.
-        thickness (int): The thickness of the slab.
-        vacuum (float): The vacuum thickness, in Angstroms.
-        xy_supercell_matrix (List[List[int]]): The supercell matrix for the xy plane.
-        use_conventional_cell (bool): Whether to use the conventional cell.
-        use_orthogonal_z (bool): Whether to use orthogonal z.
-        make_primitive (bool): Whether to try to find primitive cell for the created slab.
+        direction (AxisEnum): The direction to add vacuum along (x, y, or z).
+        size (float): Size of the vacuum gap in angstroms. Defaults to 10.0.
     """
 
+    type: str = "VacuumConfiguration"
+    direction: AxisEnum = VacuumConfigurationSchema.model_fields["direction"].default
+    size: float = VacuumConfigurationSchema.model_fields["size"].default
+
+
+class SlabConfiguration(SlabConfigurationSchema, BaseConfigurationPydantic):
     type: str = "SlabConfiguration"
-    bulk: Material
+    stack_components: List[Union[AtomicLayersUniqueRepeated, VacuumConfiguration]]
+    xy_supercell_matrix: Optional[List[List[int]]] = SlabConfigurationSchema.model_fields["xy_supercell_matrix"].default
+    direction: AxisEnum = AxisEnum.z
+
+    def __init__(self, stack_components: List[Any], **kwargs: Any):
```

Comment:
Instead of modifying `_init_` we should isolate this as `from_dict` or `from_json` (depending on what is the actual type of data stored in the metadata` and use that in the defect creation explicitly